pkg/querier/queryrange/codec.go

```diff
@@ -1962,10 +1965,27 @@ func mergeLokiResponse(responses ...queryrangebase.Response) *LokiResponse {
 		lokiResponses = make([]*LokiResponse, 0, len(responses))
 	)
 
+	uniqueWarnings := map[string]struct{}{}
 	for _, res := range responses {
 		lokiResult := res.(*LokiResponse)
 		mergedStats.MergeSplit(lokiResult.Statistics)
 		lokiResponses = append(lokiResponses, lokiResult)
+
+		for _, w := range lokiResult.Warnings {
+			uniqueWarnings[w] = struct{}{}
+		}
+	}
+
+	warnings := make([]string, 0, len(uniqueWarnings))
```

Comment:
Nit, but since [we already import `golang.org/x/exp`](https://github.com/grafana/loki/blob/b66c3433831fbd2c257427fa9a0d6ea52f3ededd/go.mod#L138), we might as well use [`maps.Keys`](https://pkg.go.dev/golang.org/x/exp/maps#Keys) from it